pcsx2/Counters.cpp

```diff
@@ -258,37 +276,52 @@ u32 UpdateVSyncRate()
 	//  the GS's output circuit.  It is the same regardless if the GS is outputting interlace
 	//  or progressive scan content. 
 
-	Fixed100	framerate = 0;
-	u32			scanlines = 0;
+	Fixed100	framerate = EmuConfig.GS.FramerateNTSC / 2;
+	u32			scanlines = SCANLINES_TOTAL_NTSC;
 	bool		isCustom  = false;
 
-	if( gsRegionMode == Region_PAL )
+	switch (gsVideoMode)
 	{
+	case GS_VideoMode::PAL:
 		isCustom = (EmuConfig.GS.FrameratePAL != 50.0);
 		framerate = EmuConfig.GS.FrameratePAL / 2;
 		scanlines = SCANLINES_TOTAL_PAL;
 		if (!gsIsInterlaced) scanlines += 3;
-	}
-	else if ( gsRegionMode == Region_NTSC )
-	{
+		break;
+
+	case GS_VideoMode::NTSC:
 		isCustom = (EmuConfig.GS.FramerateNTSC != 59.94);
 		framerate = EmuConfig.GS.FramerateNTSC / 2;
 		scanlines = SCANLINES_TOTAL_NTSC;
 		if (!gsIsInterlaced) scanlines += 1;
-	}
-	else if ( gsRegionMode == Region_NTSC_PROGRESSIVE )
-	{
+		break;
+
+	case GS_VideoMode::HDTV_480P: case GS_VideoMode::HDTV_576P:  case GS_VideoMode::VESA:
+	case GS_VideoMode::HDTV_720P: case GS_VideoMode::HDTV_1080I: case GS_VideoMode::HDTV_1080P:
 		isCustom = (EmuConfig.GS.FramerateNTSC != 59.94);
 		framerate = EmuConfig.GS.FramerateNTSC / 2;
 		scanlines = SCANLINES_TOTAL_NTSC;
+		break;
+
+	// SYSCALL instruction hasn't executed yet, nothing to do here.
+	case GS_VideoMode::Uninitialized:
+		break;
+
+	// Falls through to unknown when unidentified mode parameter of SetGSCRT is detected.
+	case GS_VideoMode::Unknown:
+		Console.Error("PCSX2-Counters: Unknown video mode detetcted");
+
+	default:
+		pxAssert(false);
 	}
 
-	if (vSyncInfo.Framerate != framerate || vSyncInfo.RegionMode != gsRegionMode)
+	if (vSyncInfo.Framerate != framerate || vSyncInfo.VideoMode != gsVideoMode)
 	{
-		vSyncInfo.RegionMode = gsRegionMode;
+		vSyncInfo.VideoMode = gsVideoMode;
 		vSyncInfoCalc( &vSyncInfo, framerate, scanlines );
-		Console.WriteLn( Color_Green, "(UpdateVSyncRate) Mode Changed to %s.", ( gsRegionMode == Region_PAL ) ? "PAL" : 
-			( gsRegionMode == Region_NTSC ) ? "NTSC" : "NTSC Progressive Scan" );
+
+		if(gsVideoMode != GS_VideoMode::Uninitialized)
+		Console.WriteLn( Color_Green, "(UpdateVSyncRate) Mode Changed to %s.", Report_VideoMode(gsVideoMode));
```

Comment:
Indentation is wrong. Also missing space after if.

`Report_VideoMode` (`ReportVideoMode`?) and `IsAnalogVideoMode` should probably be made static by the way.